packages/helpermodules/command.py

```diff
@@ -687,6 +691,48 @@ def restoreBackup(self, connection_id: str, payload: dict) -> None:
                              f'Restore-Status: {result.returncode}<br />Meldung: {result.stdout.decode("utf-8")}',
                              MessageType.ERROR)
 
+    def requestMSALAuthCode(self, connection_id: str, payload: dict) -> None:
+        """ startet den Authentifizierungsprozess für MSAL (Microsoft Authentication Library) um Onedrive Backup zu ermöglichen
+        """
+
+        # to-do: Konfiguration aus OneDrive modul importieren?
+        # Define the scope of access
+        scope = {"https://graph.microsoft.com/Files.ReadWrite"} # Replace with your desired scope
+
+        # Define the authority and the token endpoint for MSA/Live accounts
+        authority = "https://login.microsoftonline.com/consumers/"
+        clientID = "e529d8d2-3b0f-4ae4-b2ba-2d9a2bba55b2"
+
+        # Create a public client application with msal
+        app = PublicClientApplication(client_id=clientID, authority=authority)
+
+        flow = app.initiate_device_flow(scope)
```

Comment:
Ähnliches Fragestellung wie an der anderen Stelle. Auch hier muss ich Informationen in die Konfiguration des Backupmoduls schreiben.
Hintergrund: Die Authentifikation erzeugt eine URL und einen Code welcher der Benutzer aufrufen und eingeben muss um den Zugriff zu authorisieren. Diese muss ich in der Konfig-UI darstellen, Idee war diese in der Konfig zu speichern um diese Werte dann in der UI anzeigen zu können.

Diese Funktion wird während der Konfiguration aufgerufen, daher ist die Konfig noch nicht gespeichert. 
1) Wie kann ich die Defaults aus der OnedriveBackup config Klasse hier abrufen? Möchte die Werte zum Scope, Authority und clientID hier nicht im Code hinterlegen. 
2) Basierend auf den abgerufenen defaults und Ergebnissen des Flow muss ich eine Konfiguration erzeugen und diese Schreiben - sollte über ein einfaches publizieren gegen den passenden Pfad möglich sein.

Reply:
@vuffiraa72 , @LKuemmel könnt ihr mir hier helfen? Sehr gern auch schon mal die restlichen Änderungen anschauen und Feedback geben. 

Sobald alles fertig ist muss ich noch ein wenig mit den Kommentaren aufräumen, das ist mir bewusst. UI habe ich bereits vorbereitet und lokal lauffähig, das kommt das sobald die Details zum Austausch Backend/Frontend erledigt sind.

Grundsätzlich ist der Code funktionsfähig, ich hab' bereits ein paar Backups auf OneDrive erzeugt.

Reply:
Mir ist eben nicht eine andere Idee gekommen, womöglich kann ich das besser über die config UI lösen. 
Ich lasse die Erstellung des auch Codes erst zu wenn die konfig gespeichert wurde. Dann kann ich existierende Mechanismen nutzen und muss das nicht in Code duplizieren.